File: backend/src/main/java/com/leavemarker/entity/Employee.java
package com.leavemarker.entity;

import com.leavemarker.enums.EmployeeStatus;
import com.leavemarker.enums.EmploymentType;
import com.leavemarker.enums.IndianState;
import com.leavemarker.enums.Role;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "employees",
       uniqueConstraints = {
           @UniqueConstraint(columnNames = {"company_id", "employee_id"}),
           @UniqueConstraint(columnNames = {"company_id", "email"})
       })
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Column(name = "employee_id", nullable = false, length = 50)
    private String employeeId;

    @Column(nullable = false, length = 100)
    private String fullName;

    @Column(nullable = false, length = 100)
    private String email;

    @Column(nullable = false, length = 255)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Role role;

    @Column(length = 100)
    private String department;

    @Column(length = 100)
    private String jobTitle;

    @Column(nullable = false)
    private LocalDate dateOfJoining;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EmploymentType employmentType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private IndianState workLocation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manager_id")
    private Employee manager;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private EmployeeStatus status = EmployeeStatus.ACTIVE;

    @Column(length = 500)
    private String passwordResetToken;

    @Column
    private LocalDate passwordResetTokenExpiry;
}

File: backend-go/internal/models/employee.go
package models

import (
	"time"

	"github.com/leavemarker/backend-go/internal/enums"
)

// Employee represents an employee entity
type Employee struct {
	BaseModel
	CompanyID                uint                  `gorm:"not null;index" json:"companyId"`
	EmployeeID               string                `gorm:"size:50;not null" json:"employeeId"`
	FullName                 string                `gorm:"size:100;not null" json:"fullName"`
	Email                    string                `gorm:"size:100;uniqueIndex;not null" json:"email"`
	Password                 string                `gorm:"size:255;not null" json:"-"`
	Role                     enums.Role            `gorm:"size:20;not null" json:"role"`
	Department               string                `gorm:"size:100" json:"department"`
	JobTitle                 string                `gorm:"size:100" json:"jobTitle"`
	DateOfJoining            *time.Time            `gorm:"type:date" json:"dateOfJoining"`
	EmploymentType           enums.EmploymentType  `gorm:"size:20;default:FULL_TIME" json:"employmentType"`
	WorkLocation             enums.IndianState     `gorm:"size:50" json:"workLocation"`
	ManagerID                *uint                 `gorm:"index" json:"managerId"`
	Status                   enums.EmployeeStatus  `gorm:"size:20;default:ACTIVE" json:"status"`
	PasswordResetToken       *string               `gorm:"size:500" json:"-"`
	PasswordResetTokenExpiry *time.Time            `gorm:"type:date" json:"-"`

	// Relationships
	Company           Company            `gorm:"foreignKey:CompanyID" json:"-"`
	Manager           *Employee          `gorm:"foreignKey:ManagerID" json:"-"`
	DirectReports     []Employee         `gorm:"foreignKey:ManagerID" json:"-"`
	LeaveApplications []LeaveApplication `gorm:"foreignKey:EmployeeID" json:"-"`
	LeaveBalances     []LeaveBalance     `gorm:"foreignKey:EmployeeID" json:"-"`
	Attendances       []Attendance       `gorm:"foreignKey:EmployeeID" json:"-"`
}

func (Employee) TableName() string {
	return "employees"
}
